purchase_transport_mode/models/purchase_order.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+    _name = "purchase.order"
+
+    transport_mode_id = fields.Many2one("purchase.transport.mode")
+    transport_mode_status = Serialized(
+        compute="_compute_transport_mode_validation_status",
+        help="Collect and validate order details to satisfy transport mode requirements",
+        default={},
+    )
+    transport_mode_status_display = fields.Html(
+        compute="_compute_transport_mode_validation_status",
+        help="Render transport_mode_status in the UI",
+    )
+    transport_mode_status_ok = fields.Boolean(
+        compute="_compute_transport_mode_validation_status",
+        help="All transport mode requirements are satisfied",
+    )
+
+    @api.depends("transport_mode_id")
+    def _compute_transport_mode_validation_status(self):
+        for rec in self:
+            rec.transport_mode_status = rec._get_transport_mode_validation_status()
+            rec.transport_mode_status_display = (
+                rec._get_transport_mode_validation_status_display()
+            )
+            rec.transport_mode_status_ok = rec.transport_mode_status == {}
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        if self.partner_id and not self.transport_mode_id:
```

Comment:
```suggestion
        if self.partner_id:
```
otherwise if you switch from a partner w/ t-mode to another w/o you won't update the t-mode

Reply:
Also, maybe worth to lookup for the t-mode on commercial partner?